R/size.R

```diff
@@ -144,6 +145,8 @@ vec_slice <- function(x, i) {
       miss_args <- rep(list(missing_arg()), d - 1)
       eval_bare(expr(x[i, !!!miss_args] <- value))
     }
+
+    check_names_all_or_nothing(names(x))
```

Comment:
This seems like a weird place to declare this invariant. I think you should remove it.

Reply:
We have an issue for it: #171. Postponing, this doesn't affect _blob_ either.